arlo-client/src/components/Audit/RoundManagement/CreateAuditBoards.tsx

```diff
@@ -2,73 +2,44 @@ import React from 'react'
 import { H4 } from '@blueprintjs/core'
 import { Formik, Field } from 'formik'
 import styled from 'styled-components'
-import { toast } from 'react-toastify'
-import { IAuditBoard, IErrorResponse } from '../../../types'
 import { generateOptions } from '../../Atoms/Form/_helpers'
 import FormButton from '../../Atoms/Form/FormButton'
 import Select from '../../Atoms/Form/Select'
-import { api, checkAndToast } from '../../utilities'
 
 const Wrapper = styled.div`
   margin: 20px 0;
 `
 
-const CreateAuditBoards = ({
-  auditBoards,
-  electionId,
-  jurisdictionId,
-  roundId,
-  getAuditBoards,
-}: {
-  auditBoards: IAuditBoard[]
-  electionId: string
-  jurisdictionId: string
-  roundId: string
-  getAuditBoards: () => void
-}) => {
-  const submit = async ({
-    auditBoards: auditBoardsCount,
-  }: {
-    auditBoards: number
-  }) => {
-    try {
-      const body = JSON.stringify(
-        [...Array(auditBoardsCount).keys()].map(i => ({
-          name: `Audit Board #${i}`,
-        }))
-      )
-      const response: IErrorResponse = await api(
-        `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/audit-board`,
-        {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-          body,
-        }
-      )
-      // checkAndToast left here for consistency and reference but not tested since it's vestigial
-      /* istanbul ignore next */
-      if (checkAndToast(response)) return
-      getAuditBoards()
-    } catch (err) /* istanbul ignore next */ {
-      // TEST TODO
-      toast.error(err.message)
-    }
+interface IValues {
+  numAuditBoards: number
+}
+
+interface IProps {
+  createAuditBoards: (auditBoards: { name: string }[]) => Promise<boolean>
+  disabled: boolean
+}
+
+const CreateAuditBoards = ({ createAuditBoards, disabled }: IProps) => {
+  const submit = async ({ numAuditBoards }: IValues) => {
+    const boards = [...Array(numAuditBoards).keys()].map(i => ({
+      name: `Audit Board #${i}`,
+    }))
+    createAuditBoards(boards)
   }
+
   return (
     <Wrapper>
       <H4>Number of Audit Boards</H4>
       <p>
-        Select the appropriate number of audit baords based upon the personnel
+        Select the appropriate number of audit boards based upon the personnel
         available and the number of ballots assigned to your jurisdiction for
         this round of the audit. You will have the opportunity to adjust the
         number of audit boards before the next round of the audit, if another
         round is required.
       </p>
       <Formik
         enableReinitialize
-        initialValues={{ auditBoards: auditBoards.length || 1 }}
+        initialValues={{ numAuditBoards: 1 }}
```

Comment:
@jonahkagan If I'm reading this right, the result will be that once the audit boards are created and the form is disabled, it will not show the correct number of audit boards in the locked form. On rerender it will default to 1, regardless of what the value actually is. Because we see this form even when we're done using it, it needs to default to the already-submitted value if it's already submitted. That's what the `enableReinitialize` option is for.

Reply:
oh yeah i'm now seeing that is what's happening. thank you